pandaserver/taskbuffer/OraDBProxy.py

```diff
@@ -21033,9 +21033,9 @@ def updateWorkers(self, harvesterID, data, useCommit=True):
                                 # insert
                                 varMap = {}
                                 varMap[':PandaID'] = pandaID
-                                varMap[':prodSourceLabel'] = JobSpec.prodSourceLabel
+                                varMap[':prodSourceLabel'] = 'unknown'
                                 varMap[':jobStatus'] = 'failed'
-                                varMap[':attemptNr'] = 0 if JobSpec.attemptNr in [None, 'NULL', ''] else JobSpec.attemptNr
+                                varMap[':attemptNr'] = 0
```

Comment:
The error was in DBProxy updateWorkers() at the part about migrating job output reports to DB:

```
2020-10-07 21:50:43,108 panda.log.DBProxy: DEBUG    updateWorkers < HarvesterID=CERN_central_B > updateworker varMap {':PandaID': 4859384746, ':prodSourceLabel': <member 'prodSourceLabel' of 'JobSpec' objects>, ':jobStatus': 'failed', ':attemptNr': <member 'attemptNr' of 'JobSpec' objects>, ':data': None, ':timeStamp': datetime.datetime(2020, 10, 7, 21, 50, 43, 108770)}
2020-10-07 21:50:43,109 panda.log.DBProxy: DEBUG    rollback
2020-10-07 21:50:43,110 panda.log.DBProxy: DEBUG    rollback EC:Pytho Python value of type member_descriptor not supported.
2020-10-07 21:50:43,116 panda.log.DBProxy: ERROR    updateWorkers < HarvesterID=CERN_central_B > updateWorkers: NotSupportedError Python value of type member_descriptor not supported.Traceback (most recent call last):
  File "/opt/panda/lib64/python3.6/site-packages/pandaserver/taskbuffer/OraDBProxy.py", line 21042, in updateWorkers
    self.cur.execute(sqlI+comment, varMap)
  File "/opt/panda/lib64/python3.6/site-packages/pandaserver/taskbuffer/WrappedCursor.py", line 89, in execute
    ret = cur.execute(sql, varDict)
cx_Oracle.NotSupportedError: Python value of type member_descriptor not supported.
```

The problem was the "JobSpec" here is incomplete and has member_descriptor in its attributes:

```
varMap {':PandaID': 4859384746, ':prodSourceLabel': <member 'prodSourceLabel' of 'JobSpec' objects>, ':jobStatus': 'failed', ':attemptNr': <member 'attemptNr' of 'JobSpec' objects>, ':data': None, ':timeStamp': datetime.datetime(2020, 10, 7, 21, 50, 43, 108770)}
```

I fixed by replacing the attribute values with arbitrary values instead since I cannot find out how to get exact prodSourceLabel and attemptNr in updateWorkers() function.

Is there any better solution?

BTW, I have no idea about member_descriptor (even after googling it) and why it appears in JobSpec...

Reply:
Something like

sqlJAC  = "SELECT jobStatus FROM ATLAS_PANDA.jobsActive4 WHERE PandaID=:PandaID  "
->
sqlJAC  = "SELECT jobStatus,prodSourceLabel,attemptNr FROM ATLAS_PANDA.jobsActive4 WHERE PandaID=:PandaID  "


jobStatus, = resJAC
->
jobStatus,prodSourceLabel,attemptNr = resJAC


varMap[':prodSourceLabel'] = 'unknown'
->
varMap[':prodSourceLabel'] = prodSourceLabel

varMap[':attemptNr'] = 0
->
varMap[':attemptNr'] = attemptNr

Reply:
Thanks.
Implement in https://github.com/PanDAWMS/panda-server/pull/109